### PR TITLE
Remove the deprecated local parameter for _cat/shards

### PR DESCRIPTION
**Pull request by danhermann**
The cat shards API performs a `ClusterStateAction` and then an `IndicesStatsAction`. Today it accepts the `?local` parameter and passes this to the `ClusterStateAction` but this parameter has no effect on the `IndicesStatsAction`. This can be surprising because `GET _cat/shards?local` looks like it might be a completely local call but in fact it still depends on every node in the cluster.

The `local` parameter was deprecated in 7.x (#62197) and this commit removes it for 8.x.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java [hunk 1]
```diff
@@ -90,10 +88,10 @@ protected void documentation(StringBuilder sb) {
     public RestChannelConsumer doCatRequest(final RestRequest request, final NodeClient client) {
         final String[] indices = Strings.splitStringByCommaToArray(request.param("index"));
         final ClusterStateRequest clusterStateRequest = new ClusterStateRequest();
-        if (request.hasParam("local")) {
-            DEPRECATION_LOGGER.deprecate("local", LOCAL_DEPRECATED_MESSAGE);
+        // needed only in v8 to catch breaking usages and can be removed in v9
+        if (request.hasParam("local")  && Version.CURRENT.major == Version.V_7_0_0.major + 1) {
+            throw new IllegalArgumentException("parameter [local] is not supported");
```
Comment (dakrone) [inline server/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java:93]: You don't need to add this I believe, we automatically throw exceptions on unrecognized parameters:

```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "request [/_cat/shards] contains unrecognized parameter: [aoeu]"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "request [/_cat/shards] contains unrecognized parameter: [aoeu]"
  },
  "status" : 400
}
```

**Review by dakrone — COMMENTED** [review]
I left a comment related to this

**Comment by danhermann** [comment]
@elasticmachine update branch

**Review by dakrone — APPROVED** [review]
LGTM

**Comment by danhermann** [comment]
Thanks, @dakrone!

**Pull request merged**